muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -508,4 +545,132 @@ public Response findMatriculationEligibility(@QueryParam ("subjectCode") String
     return Response.ok(result).build();
   }
   
+  @GET
+  @Path("/workspaces/")
+  @RESTPermit (handling = Handling.INLINE)
+  public Response listWorkspaces(
+        @QueryParam("userIdentifier") String userIdentifier,
+        @Context Request request) {
+    List<fi.otavanopisto.muikku.plugins.workspace.rest.model.Workspace> workspaces = new ArrayList<>();
+
+    if (!sessionController.isLoggedIn()) {
```

Comment:
Maybe we could start moving all of these to...

`@RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)`